pkg/virt-launcher/virtwrap/network/common.go

```diff
@@ -509,3 +514,18 @@ func filterPodNetworkRoutes(routes []netlink.Route, nic *VIF) (filteredRoutes []
 	}
 	return
 }
+
+func retry(f func() error, attempts uint, backoffTime time.Duration) (uint, error) {
```

Comment:
What about using https://pkg.go.dev/k8s.io/client-go/util/retry#OnError ?

Reply:
I tried but it is more noisy and I still will need to add another wrapper over it.
This one is simpler, if we will need the complexity of that one we can convert to use it in the future.

Reply:
What about `wait.Poll` (or PollImmediate) ?

Example
```
	err := wait.Poll(time.Second*5, time.Minute*3, func() (bool, error) {
		_, err := clusterApi.CoreV1().Secrets(namespace).Get(secret.Name, metav1.GetOptions{})
		if err != nil {
			if errors.IsNotFound(err) {
				return true, nil
			}
			return false, nil
		}
		return false, fmt.Errorf("secret %s already exists", secret.Name)
	})

	if err != nil {
		return err
	}
```

Reply:
I did not try the wait-pool, so thank you for the suggestion. But why you are giving an unrelated example?
If you manage to show me that we can implement the existing with the `wait.Poll` and still have:
- The number of attempts returned in order to add it to the log.
- Report all errors in case of failure.

And at the same time keep it not-too-complex, then I will gladly take it.

Reply:
Since the tap creation command is does not block forever if there is a problem the `retry` implementation from @EdDev is good enough, `wait.Poll` and friends is better suited for scenarios where a call can be blocked forever I we want to denote a timeout there.

Reply:
I understand what you say Quique, but i think that even without the need of this advanced feature, it can be used
for a plain retry, without implementing new one.
It is also a poll after all.

Pseudo code for your consideration,

for example if you want 3 retries, 1 every second
it will print each error.

```
	err := wait.Poll(time.Second, 3*time.Second, func() (bool, error) {
                count++
		_, err := command()
		if err != nil {
			Log Error
                        return false, nil
		}
                print count in case its interesting also in the good flow
		return true, nil
	})

	if err != nil {
                print count (or just add some buffers to the timers so it will be fixed 3 in case of failure)
		return err
	}
```

Reply:
Your proposal has different semantics which may be fine, but the main reason not to use it in this case is that it adds nothing except requiring me to use another redirection level with no abstraction advantages.

If I will need to change something in this PR, I will try it out to see how much mess it does. Maybe that will convince me.